lib/extendOrdWithCustom.js

```diff
@@ -0,0 +1,86 @@
+const path = require("path");
+const cds = require("@sap/cds");
+const _ = require('lodash');
+
+const ORD_ENV_KEY = 'ord';
+const CONTENT_MERGE_KEY = 'ordId'
+
+function hasStringConflictsWithCdsrcJson(key) {
+    if (key in cds.env[ORD_ENV_KEY]) {
+        console.error(`Detect ord custom content has conflict with .cdsrc.json: ${key},
+            please resolve it, use .cdsrc.json value.`);
+        return true;
+    }
+    return false;
+}
```

Comment:
Not sure if we really need this. What you have in the `ord` object in .cdsrc file is not an ORD document and should not be merged anyway. We just merge the generated ORD content with the override content that is provided in the linked file.

Reply:
I want to warn users if they set top-level string property in customOrdFile, which won't be merged. Maybe give the function a better name? Or we don't need to handle this case and remove it?

Reply:
Yes, maybe the issue is then just the name of the function (which states cdsrc file as the source).

Reply:
Good point, I also have an idea that we need to set proper logger. I'm not familiar with javascript logger, and I'm surprised that there is no buildin logger package inside nodejs. I found that cds providing a [logger ](https://cap.cloud.sap/docs/node.js/cds-log). What's your opinion?